lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosValidation.cxx

```diff
@@ -0,0 +1,184 @@
+FunctionalChaosValidation::FunctionalChaosValidation(const FunctionalChaosResult & functionalChaosResult)
+  : PersistentObject()
+  , functionalChaosResult_(functionalChaosResult)
+{
+  // Nothing to do
+}
+
+
+/* Virtual constructor */
+FunctionalChaosValidation * FunctionalChaosValidation::clone() const
+{
+  return new FunctionalChaosValidation(*this);
+}
+
+/* String converter */
+String FunctionalChaosValidation::__repr__() const
+{
+  OSS oss;
+  oss << "class=" << FunctionalChaosValidation::GetClassName()
+      << " functional chaos result=" << functionalChaosResult_;
+  return oss;
+}
+
+/* Get result*/
+FunctionalChaosResult FunctionalChaosValidation::getResult() const
+{
+  return functionalChaosResult_;
+}
+
+/* Compute LOO error */
+Point FunctionalChaosValidation::computeLeaveOneOutError() const
+{
+  const FunctionCollection reducedBasis(functionalChaosResult_.getReducedBasis());
+  const Function transformation(functionalChaosResult_.getTransformation());
+  const Sample inputSample(functionalChaosResult_.getInputSample());
+  const Sample standardSample(transformation(inputSample));
+  DesignProxy designProxy(standardSample, reducedBasis);
+  const UnsignedInteger reducedBasisSize = reducedBasis.getSize();
+  Indices allIndices(reducedBasisSize);
+  allIndices.fill();
+  const MatrixImplementation designMatrix(designProxy.computeDesign(allIndices));
+  // The method name is set to the default one, given by ResourceMap
+  String methodName = ResourceMap::GetAsString("LeastSquaresExpansion-DecompositionMethod");
+  LeastSquaresMethod leastSquaresMethod = LeastSquaresMethod::Build(methodName, designProxy, allIndices);
+  const Sample outputSample(functionalChaosResult_.getOutputSample());
+  const Function metamodel(functionalChaosResult_.getMetaModel());
+  const Sample outputMetamodel(metamodel(inputSample));
+  const UnsignedInteger outputDimension = outputSample.getDimension();
+  const UnsignedInteger sampleSize = inputSample.getSize();
+  Point leaveOneOutMSE(outputDimension);
+  for (UnsignedInteger j = 0; j < outputDimension; ++j)
+  {
+    const Sample marginalOutputSample(outputSample.getMarginal(j));
+    const Point marginalOutputPoint(marginalOutputSample.asPoint());
+    // Necessary to compute the projection matrix
+    const Point betaHat = leastSquaresMethod.solve(marginalOutputPoint);
+    const Point diagonalH = leastSquaresMethod.getHDiag();
+    const Sample outputMetamodelMarginal(outputMetamodel.getMarginal(j));
+    const Point residuals = marginalOutputPoint - outputMetamodelMarginal.asPoint();
+    Point correctedResiduals(sampleSize);
+    Point squaredCorrectedResiduals(sampleSize);
+    for (UnsignedInteger i = 0; i < sampleSize; ++i)
+    {
+      correctedResiduals[i] = residuals[i] / (1.0 - diagonalH[i]);
+      squaredCorrectedResiduals[i] = correctedResiduals[i] * correctedResiduals[i];
+    }
+    const Sample squaredCorrectedResidualsSample(Sample::BuildFromPoint(squaredCorrectedResiduals));
```

Comment:
This could/should be avoid!
No need to store `squaredCorrectedResiduals`and `correctedResiduals` also !
```cxx
// in i loop
const Scalar correctedResiduals =  (residuals[i] / (1.0 - diagonalH[i]);
leaveOneOutMSE[j] += correctedResiduals * correctedResiduals;
// outside i loop
leaveOneOutMSE[j] /= sampleSize;
```

Reply:
Fixed at 4a115d52d. I removed the unnecessary `correctedResiduals` and I retained the call to `Sample.computeMean()` as it clarifies the code. I removed the unnecessary `squaredCorrectedResidualsSample`, which simplifies and speed up the code.